text/0002-numeric-values.md

```diff
@@ -0,0 +1,485 @@
+
+## Built-in Properties on the Float Type
+
+`f64` has built in properties. These are accessed by using property access notation.
+
+- `f64.min`: Constant `1.7976931348623157e+308`
+- `f64.max`: Constant `5e-324`
+- `f64.inf`: Constant `Infinity`
+
+## Built-in Properties on Float Values
+
+Similarly to integers, floats have built-in properties and methods. The variable name `val` is a placeholder for the float value used.
+
+- `val.to(T)`: Converts the float's value to a the given type. T can be
+  - An integer type like `u8`, which it will convert the value to that type, rounding the float if needed.
+    - It is safety-checked undefined behavior if the float is out of range of `T.min`..`T.max`.
+    - It is safety-checked undefined behavior if the float is not an integer.
+  - `f64`, which it will do nothing.
+- `val.round()`: Round this float to the nearest integer, while keeping it a float.
+- `val.floor()`: Round this float down to the nearest integer, while keeping it a float.
+- `val.ceil()`: Round this float up to the nearest integer, while keeping it a float.
+- `val.pow(y)`: Raise this float to the power of `y`.
+  - It is safety-checked undefined behavior to call `1.pow(f64.inf)`.
+- `val.sqrt()`: Get the square root of this float.
+  - It is safety-checked undefined behavior if the value is negative.
+- `val.abs()`: Get the absolute value of this float.
+
+## Float Arithmetic
+
+The following binary operators seen in most other languages are available:
+
+- `+`: Addition
+- `-`: Subtraction
+- `*`: Multiplication
+- `/`: Division
+- `%`: Remainder
+
+The following unary operators are available:
+
+- `-`: Negation
+
+The following operations will cause undefined behavior:
+
+- `/`: If the divisor is `0`.
+- `%`: If the divisor is `0`.
+
+Note that floating point operations cannot overflow, but instead will return `f64.inf` as IEEE 754 specifies.
+
+## NaN
+
+There is no `NaN` constant. Instead, any operation that could produce `NaN` is either safety-checked undefined behavior (like dividing by zero), or will return an optional `f64`. When representing `NaN` is required, it should be represented as an optional `f64`, to indicate the possibility of `NaN`.
+
+If it was missed when writing this RFC. Any math operation that would produce `NaN` should be safety-checked undefined behavior.
+
+> [!NOTE]
+> There is not yet an RFC for optionals.
+
+# Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+## Parsing numbers
+
+For reference, the following regular expression is a set of all valid numeric literals:
+
+```regexp
+^(?:\d(?:\d|_\d)*)?\.?(?:\d(?:\d|_\d)*)+(?:e[+-]?(?:\d(?:\d|_\d)*))?\b
+```
+
+A compiler should implement this in native code than using a regular expression engine.
+
+Examples of valid numeric literals:
+
+- `10` (an integer)
+- `512_000.000` (an integer)
+- `63_2.42` (a float)
+- `.425_000` (a float)
```

Comment:
this one feels a bit dubious to me

Reply:
leading . is usually valid in most languages. _ separators are also featured in some. this test case is two of those at once.